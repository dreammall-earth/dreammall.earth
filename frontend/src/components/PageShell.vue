<template>
  <v-app>
    <slot />
  </v-app>
</template>

<script lang="ts" setup>
import TopMenu from '#components/menu/TopMenu.vue'
</script>
