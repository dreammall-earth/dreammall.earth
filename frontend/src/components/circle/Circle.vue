<template>
  <div
    class="circle rounded-circle border-sm text-icon bg-icon-background pa-2 d-flex justify-center align-center"
  >
    <slot />
  </div>
</template>

<style scoped lang="scss">
.circle {
  width: var(--menu-icon-height);
  height: var(--menu-icon-height);
}
</style>
