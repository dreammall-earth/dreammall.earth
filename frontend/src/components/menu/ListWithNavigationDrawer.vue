<template>
  <NavigationDrawer :modelValue="drawer" @update:modelValue="updateDrawer">
    <ListElement :items="items" />
  </NavigationDrawer>
</template>

<script lang="ts" setup>
import { ref, defineProps, defineEmits } from 'vue'
import NavigationDrawer from './NavigationDrawer.vue'
import ListElement from './ListElement.vue'

const props = defineProps({
  drawer: {
    type: Boolean,
    required: true
  }
})

const emits = defineEmits(['update:drawer'])

const items = ref([
  { title: 'LOLLY Krypto Entwicklung', fullWidth: false },
  { title: 'Aachener Freunde Treff', fullWidth: false },
  { title: 'Building A Wooden Guitar', fullWidth: false },
  { title: 'Co Working Space Manufaktur', fullWidth: true },
  { title: 'Intervallfasten Kalender', fullWidth: false },
  { title: 'Lomografie 2050', fullWidth: true },
  { title: 'Freies Treffen U50', fullWidth: false },
])

const updateDrawer = (value: boolean) => {
  emits('update:drawer', value)
}
</script>

<style scoped>
.list-container {
  margin-top: 20px;
}
</style>

