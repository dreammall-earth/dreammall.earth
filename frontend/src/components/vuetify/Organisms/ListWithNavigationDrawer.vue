<template>
    <NavigationDrawer :model-value="drawer" :location="location" @update:model-value="updateDrawer">
      <SearchField
        v-model="search"
        label="Open Tables, Jobs"
        prepend-inner-icon="mdi-tune"
      ></SearchField>
      <v-list>
        <div class="mx-4">{{ $t('menu.roomList') }}</div>
        <ListElement :items="filteredItems" @item-click="updateDrawer(false)" />
      </v-list>
    </NavigationDrawer>
  </template>
  
  <script lang="ts" setup>
  import { ref, computed } from 'vue'
  import { VAvatar, VIcon, VImg } from 'vuetify/components'
  import SearchField from '../search/SearchField.vue'
  import ListElement from './../Atoms/ListElement.vue'
  import NavigationDrawer from './../Atoms/NavigationDrawer.vue'
  defineProps({
    drawer: {
      type: Boolean,
      required: true,
    },
    location: {
      type: String as () => 'right' | 'bottom' | 'left' | 'end' | 'top' | 'start' | undefined,
      required: false,
      default: 'right',
    },
  })
  const emits = defineEmits(['update:drawer'])
  const search = ref('')
  const items = [
    {
      title: 'LOLLY Krypto Entwicklung',
      subtitle: 'Krypto fÃ¼r Alle',
      prepend: VAvatar,
      prependProps: { color: 'primary', icon: 'mdi-lock' },
      append: VIcon,
      appendProps: { icon: 'mdi-dots-vertical', class: 'append-icon' },
    },
    {
      title: 'Aachener Freunde Treff',
      subtitle: 'Aachener Freunde Treff',
      prepend: VImg,
      prependProps: {
        src: 'https://picsum.photos/40',
        referrerpolicy: 'no-referrer',
        width: 40,
        height: 40,
      },
      append: VIcon,
      appendProps: { icon: 'mdi-dots-vertical', class: 'append-icon' },
    },
    {
      title: 'Building A Wooden Guitar',
      subtitle: 'Building A Wooden Guitar',
      prepend: VImg,
      prependProps: {
        src: 'https://picsum.photos/39',
        referrerpolicy: 'no-referrer',
        width: 40,
        height: 40,
      },
      append: VIcon,
      appendProps: { icon: 'mdi-dots-vertical', class: 'append-icon' },
    },
    {
      title: 'Co Working Space Manufaktur',
      subtitle: 'Co Working Space Manufaktur',
      prepend: VImg,
      prependProps: {
        src: 'https://picsum.photos/41',
        referrerpolicy: 'no-referrer',
        width: 40,
        height: 40,
      },
      append: VIcon,
      appendProps: { icon: 'mdi-dots-vertical', class: 'append-icon' },
    },
    {
      title: 'Intervallfasten Kalender',
      subtitle: 'Intervallfasten Kalender',
      prepend: VImg,
      prependProps: {
        src: 'https://picsum.photos/42',
        referrerpolicy: 'no-referrer',
        width: 40,
        height: 40,
      },
      append: VIcon,
      appendProps: { icon: 'mdi-dots-vertical-circle', class: 'append-icon' },
    },
    {
      title: 'Lomografie 2050',
      subtitle: 'Lomografie 2050',
      prepend: VImg,
      prependProps: {
        src: 'https://picsum.photos/46',
        referrerpolicy: 'no-referrer',
        width: 40,
        height: 40,
      },
      append: VIcon,
      appendProps: { icon: 'mdi-dots-vertical-circle', class: 'append-icon' },
    },
    {
      title: 'Freies Treffen U50',
      subtitle: 'Freies Treffen U50',
      prepend: VAvatar,
      prependProps: { color: 'primary', image: 'https://picsum.photos/46' },
      append: VIcon,
      appendProps: { icon: 'mdi-dots-vertical', class: 'append-icon' },
    },
  ]
  const filteredItems = computed(() => {
    if (!search.value) {
      return items
    }
    return items.filter((item) => item.title.toLowerCase().includes(search.value.toLowerCase()))
  })
  const updateDrawer = (value: boolean) => {
    emits('update:drawer', value)
  }
  </script>