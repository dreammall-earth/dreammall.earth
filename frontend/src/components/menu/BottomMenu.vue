<template>
  <div  class="bottom-menu d-flex w-100 position-fixed bottom-0 justify-space-around align-center py-2 bg-surface">
 
        <MessageIndicator :number-of-messages="3" />
        <NewsIndicator :has-news="true" />
     
        <Circle @click="toggleDrawer">
          <v-icon icon="$camera"></v-icon>
        </Circle>
      
      <UserInfo />
  
   
    <ListWithNavigationDrawer
      :drawer="drawer"
      :location="location"
      @update:drawer="drawer = $event"
    />
  </div>
</template>

<script lang="ts" setup>
import { ref } from 'vue'

import Circle from './CircleElement.vue'
import ListWithNavigationDrawer from './ListWithNavigationDrawer.vue'
import MessageIndicator from './MessageIndicator.vue'
import NewsIndicator from './NewsIndicator.vue'
import UserInfo from './UserInfo.vue'

const drawer = ref(false)

const toggleDrawer = () => {
  drawer.value = !drawer.value
}

const location = ref<'bottom' | 'right' | 'left' | 'end' | 'top' | 'start'>('bottom')
</script>

<style scoped lang="scss">
/*v-bottom-navigation {
  z-index: 10;
}*/
.bottom-menu {
  bottom: 0;
  left: 0;
  background: var(--v-bottom-menu-background) !important;
  backdrop-filter: blur(20px);
  border-radius: 30px 30px 0 0;
}
</style>
