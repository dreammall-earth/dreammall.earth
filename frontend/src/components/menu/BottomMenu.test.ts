import { flushPromises, mount, VueWrapper } from '@vue/test-utils'
import { describe, it, expect, beforeEach, vi } from 'vitest'
import { h } from 'vue'
import { VApp } from 'vuetify/components'

import BottomMenu from './BottomMenu.vue'
import Circle from './CircleElement.vue'
import ListWithNavigationDrawer from './ListWithNavigationDrawer.vue'

describe('BottomMenu', () => {
  let wrapper: VueWrapper<any>
  let toggleDrawer: ReturnType<typeof vi.fn>

  const Wrapper = () => {
    return mount(VApp, {
      slots: {
        default: h(BottomMenu),
      },
    })
  }

  beforeEach(() => {
    toggleDrawer = vi.fn()
    wrapper = mount(BottomMenu, {
      global: {
        mocks: {
          toggleDrawer,
        },
      },
    })
  })

  it('renders BottomMenu', () => {
    expect(wrapper.element).toMatchSnapshot()
  })

  it('toggles drawer on Circle click', async () => {
    await wrapper.findComponent(Circle).trigger('click')
    expect(toggleDrawer).toHaveBeenCalled()
    await flushPromises()
    expect(wrapper.findComponent(ListWithNavigationDrawer).props('drawer')).toBe(true)
  })

  it('passes location prop to ListWithNavigationDrawer', () => {
    const listWithNavigationDrawer = wrapper.findComponent(ListWithNavigationDrawer)
    expect(listWithNavigationDrawer.props('location')).toBe('bottom')
  })

  describe('signout button', () => {
    // Mocking authStore and authService
    const authStore = {
      save: vi.fn(),
      clear: vi.fn(),
    }

    const authService = {
      signOut: vi.fn().mockResolvedValue('signed out'),
    }

    const authServiceSpy = vi.spyOn(authService, 'signOut')
    const storeSpy = vi.spyOn(authStore, 'clear')

    beforeEach(() => {
      vi.clearAllMocks()
      authStore.save({
        access_token: 'access_token',
        profile: {
          aud: 'aud',
          sub: 'sub',
          exp: 1,
          iat: 1,
          iss: 'iss',
        },
        token_type: 'token_type',
        session_state: null,
        state: null,
        expires_at: new Date().valueOf() + 100,
        expires_in: 0,
        expired: false,
        scopes: ['email'],
        toStorageString: () => 'toStorageString',
      })
    })

    describe('without error', () => {
      beforeEach(async () => {
        await wrapper.find('button.user-info').trigger('click')
        await flushPromises()
        await wrapper.find('button.sign-out').trigger('click')
      })

      it('calls auth service sign out', () => {
        expect(authServiceSpy).toBeCalled()
      })

      it('clears the store', () => {
        expect(storeSpy).toBeCalled()
      })
    })

    describe('with error', () => {
      const consoleSpy = vi.spyOn(console, 'log')

      beforeEach(async () => {
        authServiceSpy.mockRejectedValue('Error!')
        await wrapper.find('button.user-info').trigger('click')
        await flushPromises()
        await wrapper.find('button.sign-out').trigger('click')
      })

      it('logs the error', () => {
        expect(consoleSpy).toBeCalledWith('auth error', 'Error!')
      })
    })
  })
})
