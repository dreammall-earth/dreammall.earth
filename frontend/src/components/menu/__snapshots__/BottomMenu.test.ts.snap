// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`BottomMenu > renders BottomMenu 1`] = `
<div
  class="v-application v-theme--light v-layout v-layout--full-height v-locale--is-ltr"
>
  <div
    class="v-application__wrap"
  >
    
    
    <div
      class="navigation-drawer-box d-md-none position-fixed mb-5 pb-5"
      data-v-96a196cc=""
    >
      
      <nav
        class="v-navigation-drawer v-navigation-drawer--bottom v-navigation-drawer--temporary v-theme--light v-navigation-drawer--mobile menu-drawer"
        data-v-cf2c9067=""
        style="bottom: 0px; z-index: 1004; transform: translateY(110%); position: fixed; transition: none !important; left: 0px; width: calc(100% - 0px - 0px);"
      >
        <!---->
        <!---->
        <div
          class="v-navigation-drawer__content"
        >
          
          <div
            class="v-input v-input--horizontal v-input--center-affix v-input--density-default v-theme--light v-locale--is-ltr v-text-field mx-4 mt-4 search"
            data-v-cf2c9067=""
          >
            <!---->
            <div
              class="v-input__control"
            >
              
              <div
                class="v-field v-field--appended v-field--center-affix v-field--prepended v-field--variant-solo v-theme--light v-field--rounded v-locale--is-ltr"
              >
                <div
                  class="v-field__overlay"
                />
                <div
                  class="v-field__loader"
                >
                  <div
                    aria-hidden="true"
                    aria-valuemax="100"
                    aria-valuemin="0"
                    class="v-progress-linear v-theme--light v-locale--is-ltr"
                    role="progressbar"
                    style="top: 0px; height: 0px; --v-progress-linear-height: 2px;"
                  >
                    <!---->
                    <div
                      class="v-progress-linear__background"
                      style="width: 100%;"
                    />
                    <transition-stub
                      appear="false"
                      css="true"
                      name="fade-transition"
                      persisted="false"
                    >
                      <div
                        class="v-progress-linear__indeterminate"
                      >
                        
                        <div
                          class="v-progress-linear__indeterminate long"
                        />
                        <div
                          class="v-progress-linear__indeterminate short"
                        />
                        
                      </div>
                    </transition-stub>
                    <!---->
                  </div>
                </div>
                <div
                  class="v-field__prepend-inner"
                >
                  <i
                    aria-hidden="true"
                    class="mdi-tune mdi v-icon notranslate v-theme--light v-icon--size-default"
                  />
                  <!---->
                </div>
                <div
                  class="v-field__field"
                  data-no-activator=""
                >
                  <label
                    aria-hidden="true"
                    class="v-label v-field-label v-field-label--floating"
                    for="input-1"
                  >
                    <!---->
                    
                    $t('tablesDrawer.searchPlaceholder')
                    
                  </label>
                  <label
                    class="v-label v-field-label"
                    for="input-1"
                  >
                    <!---->
                    
                    $t('tablesDrawer.searchPlaceholder')
                    
                  </label>
                  
                  
                  <!---->
                  <input
                    aria-describedby="input-1-messages"
                    class="v-field__input"
                    id="input-1"
                    size="1"
                    type="text"
                  />
                  <!---->
                  
                  
                </div>
                <transition-stub
                  appear="false"
                  css="true"
                  name="expand-x-transition"
                  persisted="false"
                >
                  <div
                    class="v-field__clearable"
                    style="display: none;"
                  >
                    
                    <i
                      aria-hidden="false"
                      aria-label="$vuetify.input.clear"
                      class="mdi-close-circle mdi v-icon notranslate v-theme--light v-icon--size-default v-icon--clickable"
                      role="button"
                      tabindex="0"
                    />
                    
                  </div>
                </transition-stub>
                <div
                  class="v-field__append-inner"
                >
                  <!---->
                  <!---->
                </div>
                <div
                  class="v-field__outline"
                >
                  <!---->
                  <!---->
                </div>
              </div>
              
            </div>
            <!---->
            <div
              class="v-input__details"
            >
              <transition-group-stub
                appear="false"
                aria-live="polite"
                class="v-messages"
                css="true"
                id="input-1-messages"
                name="slide-y-transition"
                persisted="false"
                role="alert"
                tag="div"
              >
                <!---->
              </transition-group-stub>
              <!---->
            </div>
          </div>
          <div
            class="v-list v-theme--light bg-transparent v-list--density-default v-list--one-line"
            data-v-cf2c9067=""
            role="listbox"
            tabindex="0"
          >
            
            <h2
              class="mx-4"
              data-v-cf2c9067=""
            >
              $t('tablesDrawer.header')
            </h2>
            <div
              class="v-card v-theme--light v-card--density-default v-card--variant-elevated mx-auto"
              data-v-cf2c9067=""
              data-v-de37061b=""
              style="width: 400px;"
            >
              <!---->
              <div
                class="v-card__loader"
              >
                <div
                  aria-hidden="true"
                  aria-valuemax="100"
                  aria-valuemin="0"
                  class="v-progress-linear v-theme--light v-locale--is-ltr"
                  role="progressbar"
                  style="top: 0px; height: 0px; --v-progress-linear-height: 2px;"
                >
                  <!---->
                  <div
                    class="v-progress-linear__background"
                    style="width: 100%;"
                  />
                  <transition-stub
                    appear="false"
                    css="true"
                    name="fade-transition"
                    persisted="false"
                  >
                    <div
                      class="v-progress-linear__indeterminate"
                    >
                      
                      <div
                        class="v-progress-linear__indeterminate long"
                      />
                      <div
                        class="v-progress-linear__indeterminate short"
                      />
                      
                    </div>
                  </transition-stub>
                  <!---->
                </div>
              </div>
              <!---->
              <!---->
              
              <div
                class="v-list v-theme--light bg-transparent v-list--density-default v-list--two-line"
                data-v-de37061b=""
                role="listbox"
                tabindex="0"
              >
                
                
                
                
              </div>
              
              <!---->
              
              <!---->
              <span
                class="v-card__underlay"
              />
              
            </div>
            
          </div>
          
        </div>
        <!---->
      </nav>
      <transition-stub
        appear="false"
        css="true"
        name="fade-transition"
        persisted="false"
      >
        <!---->
      </transition-stub>
      
    </div>
    <div
      class="bottom-menu d-flex w-100 position-fixed bottom-0 justify-space-around align-center py-2 bg-surface d-md-none"
      data-v-96a196cc=""
    >
      <div
        class="circle rounded-circle border-sm text-icon pa-2 d-flex justify-center align-center camera-button"
        data-v-39e8094e=""
        data-v-96a196cc=""
      >
        
        <i
          aria-hidden="true"
          class="v-icon notranslate v-theme--light v-icon--size-default"
          data-v-96a196cc=""
        >
          <svg
            fill="none"
            height="17"
            viewBox="0 0 21 17"
            width="21"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M7.68844 12.4824H9.66834V9.51172H12.6382V7.53125H9.66834V4.56055H7.68844V7.53125H4.71859V9.51172H7.68844V12.4824ZM2.73869 16.4434C2.19422 16.4434 1.72812 16.2494 1.34038 15.8616C0.952654 15.4738 0.758789 15.0075 0.758789 14.4629V2.58008C0.758789 2.03545 0.952654 1.56921 1.34038 1.18137C1.72812 0.79353 2.19422 0.599609 2.73869 0.599609H14.6181C15.1626 0.599609 15.6287 0.79353 16.0164 1.18137C16.4041 1.56921 16.598 2.03545 16.598 2.58008V7.03613L20.5578 3.0752V13.9678L16.598 10.0068V14.4629C16.598 15.0075 16.4041 15.4738 16.0164 15.8616C15.6287 16.2494 15.1626 16.4434 14.6181 16.4434H2.73869ZM2.73869 14.4629H14.6181V2.58008H2.73869V14.4629Z"
              fill="currentColor"
            />
          </svg>
        </i>
        
      </div>
      <div
        class="create-button-container"
        data-v-1c60eb53=""
        data-v-96a196cc=""
      >
        <div
          data-v-1c60eb53=""
        >
          <div
            class="button-wrapper"
            data-v-1c60eb53=""
          >
            <!--teleport start-->
            <!--teleport end-->
            <svg
              data-v-1c60eb53=""
              fill="none"
              height="182"
              viewBox="0 0 182 182"
              width="182"
              xmlns="http://www.w3.org/2000/svg"
            >
              <g
                class="most-outer-rings"
                data-v-1c60eb53=""
                filter="url(#filter0_i_1513_6638)"
                opacity="0.25"
              >
                <path
                  d="M91 170.364C46.7808 170.364 11 134.653 11 90.6818C11 46.7111 46.7808 11 91 11C135.219 11 171 46.7111 171 90.6818C171 134.653 135.219 170.364 91 170.364Z"
                  data-v-1c60eb53=""
                  stroke="url(#paint0_linear_1513_6638)"
                  stroke-width="22"
                />
              </g>
              <g
                class="outer-rings"
                data-v-1c60eb53=""
                filter="url(#filter1_i_1513_6638)"
              >
                <circle
                  cx="91.3183"
                  cy="90.6817"
                  data-v-1c60eb53=""
                  r="61.5"
                  stroke="url(#paint1_linear_1513_6638)"
                  stroke-opacity="0.5"
                  stroke-width="24"
                  transform="rotate(90 91.3183 90.6817)"
                />
              </g>
              <g
                data-v-1c60eb53=""
                filter="url(#filter2_i_1513_6638)"
              >
                <circle
                  cx="91.1363"
                  cy="90.7914"
                  data-v-1c60eb53=""
                  fill="#EEEFF0"
                  fill-opacity="0.0405"
                  r="59.9955"
                  transform="rotate(90 91.1363 90.7914)"
                />
              </g>
              <circle
                cx="91.1363"
                cy="90.7914"
                data-v-1c60eb53=""
                fill="white"
                r="49.9955"
                stroke="url(#paint2_linear_1513_6638)"
                stroke-width="20"
                transform="rotate(90 91.1363 90.7914)"
              />
              <g
                clip-path="url(#clip0_1513_6638)"
                data-v-1c60eb53=""
                id="create-button-mobile"
              >
                <rect
                  data-v-1c60eb53=""
                  fill="transparent"
                  height="100%"
                  width="100%"
                />
                <path
                  d="M90.7724 80.0176H95C100.877 80.0176 105.727 84.7568 105.727 90.5C105.727 96.2441 100.877 100.982 95 100.982H90.7714H84.2733H80.5162H76.7581H73V112H95C107.101 112 117 102.325 117 90.5C117 78.6747 107.101 69 95 69H73V85.1057V87.3626H76.7581H80.5162H84.2743H88.0324H91.7905H95C96.7284 87.3626 98.2105 88.8109 98.2105 90.5C98.2105 92.1891 96.7284 93.6374 95 93.6374H91.7895H73V97.3099H88.0324H95C98.8027 97.3099 101.969 94.217 101.969 90.5C101.969 86.7839 98.8027 83.6901 95 83.6901H88.0314H84.2733H80.5162H76.7581V72.6725H80.5162H95C105.026 72.6725 113.243 80.7027 113.243 90.5C113.243 100.298 105.027 108.327 95 108.327H76.7581V104.655H80.5162H90.7724H95C102.951 104.655 109.485 98.2712 109.485 90.5C109.485 82.7297 102.951 76.346 95 76.346H90.7714H81.617V80.0176H90.7724Z"
                  data-v-1c60eb53=""
                  fill="url(#paint3_linear_1513_6638)"
                />
                <path
                  d="M81.537 87H93.9213H95V84.0007H93.9213L86.0749 84H81.537H77V87H81.537Z"
                  data-v-1c60eb53=""
                  fill="url(#paint4_linear_1513_6638)"
                  opacity="0.75"
                />
                <path
                  d="M81.6397 104.88H94.304H95.407V101.001H94.304L86.2803 101H81.6397H77V104.88H81.6397Z"
                  data-v-1c60eb53=""
                  fill="url(#paint5_linear_1513_6638)"
                  opacity="0.75"
                />
              </g>
              <defs
                data-v-1c60eb53=""
              >
                <filter
                  color-interpolation-filters="sRGB"
                  data-v-1c60eb53=""
                  filterUnits="userSpaceOnUse"
                  height="183.364"
                  id="filter0_i_1513_6638"
                  width="182"
                  x="0"
                  y="0"
                >
                  <feflood
                    data-v-1c60eb53=""
                    flood-opacity="0"
                    result="BackgroundImageFix"
                  />
                  <feblend
                    data-v-1c60eb53=""
                    in="SourceGraphic"
                    in2="BackgroundImageFix"
                    mode="normal"
                    result="shape"
                  />
                  <fecolormatrix
                    data-v-1c60eb53=""
                    in="SourceAlpha"
                    result="hardAlpha"
                    type="matrix"
                    values="0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 127 0"
                  />
                  <feoffset
                    data-v-1c60eb53=""
                    dy="2"
                  />
                  <fegaussianblur
                    data-v-1c60eb53=""
                    stdDeviation="2.5"
                  />
                  <fecomposite
                    data-v-1c60eb53=""
                    in2="hardAlpha"
                    k2="-1"
                    k3="1"
                    operator="arithmetic"
                  />
                  <fecolormatrix
                    data-v-1c60eb53=""
                    type="matrix"
                    values="0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0.25 0"
                  />
                  <feblend
                    data-v-1c60eb53=""
                    in2="shape"
                    mode="normal"
                    result="effect1_innerShadow_1513_6638"
                  />
                </filter>
                <filter
                  color-interpolation-filters="sRGB"
                  data-v-1c60eb53=""
                  filterUnits="userSpaceOnUse"
                  height="149"
                  id="filter1_i_1513_6638"
                  width="147"
                  x="17.8184"
                  y="17.1816"
                >
                  <feflood
                    data-v-1c60eb53=""
                    flood-opacity="0"
                    result="BackgroundImageFix"
                  />
                  <feblend
                    data-v-1c60eb53=""
                    in="SourceGraphic"
                    in2="BackgroundImageFix"
                    mode="normal"
                    result="shape"
                  />
                  <fecolormatrix
                    data-v-1c60eb53=""
                    in="SourceAlpha"
                    result="hardAlpha"
                    type="matrix"
                    values="0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 127 0"
                  />
                  <feoffset
                    data-v-1c60eb53=""
                    dy="2"
                  />
                  <fegaussianblur
                    data-v-1c60eb53=""
                    stdDeviation="2.5"
                  />
                  <fecomposite
                    data-v-1c60eb53=""
                    in2="hardAlpha"
                    k2="-1"
                    k3="1"
                    operator="arithmetic"
                  />
                  <fecolormatrix
                    data-v-1c60eb53=""
                    type="matrix"
                    values="0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0.25 0"
                  />
                  <feblend
                    data-v-1c60eb53=""
                    in2="shape"
                    mode="normal"
                    result="effect1_innerShadow_1513_6638"
                  />
                </filter>
                <filter
                  color-interpolation-filters="sRGB"
                  data-v-1c60eb53=""
                  filterUnits="userSpaceOnUse"
                  height="121.991"
                  id="filter2_i_1513_6638"
                  width="119.991"
                  x="31.1406"
                  y="30.7959"
                >
                  <feflood
                    data-v-1c60eb53=""
                    flood-opacity="0"
                    result="BackgroundImageFix"
                  />
                  <feblend
                    data-v-1c60eb53=""
                    in="SourceGraphic"
                    in2="BackgroundImageFix"
                    mode="normal"
                    result="shape"
                  />
                  <fecolormatrix
                    data-v-1c60eb53=""
                    in="SourceAlpha"
                    result="hardAlpha"
                    type="matrix"
                    values="0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 127 0"
                  />
                  <feoffset
                    data-v-1c60eb53=""
                    dy="2"
                  />
                  <fegaussianblur
                    data-v-1c60eb53=""
                    stdDeviation="2.5"
                  />
                  <fecomposite
                    data-v-1c60eb53=""
                    in2="hardAlpha"
                    k2="-1"
                    k3="1"
                    operator="arithmetic"
                  />
                  <fecolormatrix
                    data-v-1c60eb53=""
                    type="matrix"
                    values="0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0.25 0"
                  />
                  <feblend
                    data-v-1c60eb53=""
                    in2="shape"
                    mode="normal"
                    result="effect1_innerShadow_1513_6638"
                  />
                </filter>
                <lineargradient
                  data-v-1c60eb53=""
                  gradientUnits="userSpaceOnUse"
                  id="paint0_linear_1513_6638"
                  x1="198.763"
                  x2="-7.6048"
                  y1="23.5654"
                  y2="168.62"
                >
                  <stop
                    data-v-1c60eb53=""
                    offset="0.0149795"
                    stop-color="#2CA5B1"
                  />
                  <stop
                    data-v-1c60eb53=""
                    offset="1"
                    stop-color="#999999"
                  />
                </lineargradient>
                <lineargradient
                  data-v-1c60eb53=""
                  gradientUnits="userSpaceOnUse"
                  id="paint1_linear_1513_6638"
                  x1="36.9187"
                  x2="153.938"
                  y1="3.64216"
                  y2="170.71"
                >
                  <stop
                    data-v-1c60eb53=""
                    offset="0.0149795"
                    stop-color="#2CA5B1"
                  />
                  <stop
                    data-v-1c60eb53=""
                    offset="1"
                    stop-color="#999999"
                  />
                </lineargradient>
                <lineargradient
                  data-v-1c60eb53=""
                  gradientUnits="userSpaceOnUse"
                  id="paint2_linear_1513_6638"
                  x1="46.7317"
                  x2="142.251"
                  y1="19.7441"
                  y2="156.116"
                >
                  <stop
                    data-v-1c60eb53=""
                    offset="0.0149795"
                    stop-color="#2CA5B1"
                  />
                  <stop
                    data-v-1c60eb53=""
                    offset="1"
                    stop-color="#999999"
                  />
                </lineargradient>
                <lineargradient
                  data-v-1c60eb53=""
                  gradientUnits="userSpaceOnUse"
                  id="paint3_linear_1513_6638"
                  x1="105.448"
                  x2="75.1548"
                  y1="116.071"
                  y2="62.3785"
                >
                  <stop
                    data-v-1c60eb53=""
                    offset="0.1721"
                    stop-color="#2CA5B1"
                  />
                  <stop
                    data-v-1c60eb53=""
                    offset="0.804"
                    stop-color="#F09630"
                  />
                </lineargradient>
                <lineargradient
                  data-v-1c60eb53=""
                  gradientUnits="userSpaceOnUse"
                  id="paint4_linear_1513_6638"
                  x1="94.9997"
                  x2="76.9996"
                  y1="85.5001"
                  y2="85.5001"
                >
                  <stop
                    data-v-1c60eb53=""
                    stop-color="white"
                    stop-opacity="0"
                  />
                  <stop
                    data-v-1c60eb53=""
                    offset="0.0405"
                    stop-color="#EEEFF0"
                    stop-opacity="0.0405"
                  />
                  <stop
                    data-v-1c60eb53=""
                    offset="0.1632"
                    stop-color="#BFC3C7"
                    stop-opacity="0.1632"
                  />
                  <stop
                    data-v-1c60eb53=""
                    offset="0.29"
                    stop-color="#979DA3"
                    stop-opacity="0.29"
                  />
                  <stop
                    data-v-1c60eb53=""
                    offset="0.4195"
                    stop-color="#777E86"
                    stop-opacity="0.4195"
                  />
                  <stop
                    data-v-1c60eb53=""
                    offset="0.5522"
                    stop-color="#5D6670"
                    stop-opacity="0.5522"
                  />
                  <stop
                    data-v-1c60eb53=""
                    offset="0.6896"
                    stop-color="#4B5460"
                    stop-opacity="0.6896"
                  />
                  <stop
                    data-v-1c60eb53=""
                    offset="0.8346"
                    stop-color="#404A56"
                    stop-opacity="0.8346"
                  />
                  <stop
                    data-v-1c60eb53=""
                    offset="1"
                    stop-color="#3D4753"
                  />
                </lineargradient>
                <lineargradient
                  data-v-1c60eb53=""
                  gradientUnits="userSpaceOnUse"
                  id="paint5_linear_1513_6638"
                  x1="95.4068"
                  x2="76.9995"
                  y1="102.94"
                  y2="102.94"
                >
                  <stop
                    data-v-1c60eb53=""
                    stop-color="white"
                    stop-opacity="0"
                  />
                  <stop
                    data-v-1c60eb53=""
                    offset="0.0405"
                    stop-color="#EEEFF0"
                    stop-opacity="0.0405"
                  />
                  <stop
                    data-v-1c60eb53=""
                    offset="0.1632"
                    stop-color="#BFC3C7"
                    stop-opacity="0.1632"
                  />
                  <stop
                    data-v-1c60eb53=""
                    offset="0.29"
                    stop-color="#979DA3"
                    stop-opacity="0.29"
                  />
                  <stop
                    data-v-1c60eb53=""
                    offset="0.4195"
                    stop-color="#777E86"
                    stop-opacity="0.4195"
                  />
                  <stop
                    data-v-1c60eb53=""
                    offset="0.5522"
                    stop-color="#5D6670"
                    stop-opacity="0.5522"
                  />
                  <stop
                    data-v-1c60eb53=""
                    offset="0.6896"
                    stop-color="#4B5460"
                    stop-opacity="0.6896"
                  />
                  <stop
                    data-v-1c60eb53=""
                    offset="0.8346"
                    stop-color="#404A56"
                    stop-opacity="0.8346"
                  />
                  <stop
                    data-v-1c60eb53=""
                    offset="1"
                    stop-color="#3D4753"
                  />
                </lineargradient>
                <clippath
                  data-v-1c60eb53=""
                  id="clip0_1513_6638"
                >
                  <rect
                    data-v-1c60eb53=""
                    fill="white"
                    height="48"
                    transform="translate(69 66)"
                    width="48"
                  />
                </clippath>
              </defs>
            </svg>
          </div>
        </div>
      </div>
      
      
      <button
        aria-expanded="false"
        aria-haspopup="menu"
        aria-owns="v-menu-3"
        class="user-info rounded-pill d-flex flex-row text-icon border-sm align-center justify-center"
        data-v-607bdd70=""
        targetref="[object Object]"
      >
        <div
          class="v-avatar v-theme--light v-avatar--density-default v-avatar--variant-flat avatar d-flex align-center text-font border-sm bg-primary"
          data-v-607bdd70=""
          style="width: 44px; height: 44px;"
        >
          
          
          <span
            data-v-607bdd70=""
          />
          
          
          
          <!---->
          <span
            class="v-avatar__underlay"
          />
          
        </div>
        <div
          class="d-flex flex-column justify-center text-right pa-1 pl-3 w-100"
          data-v-607bdd70=""
        >
          <i
            aria-hidden="true"
            class="v-icon notranslate v-theme--light v-icon--size-default ellipsis-icon"
            data-test="user-dropdown"
            data-v-607bdd70=""
          >
            <svg
              fill="none"
              height="4"
              viewBox="0 0 18 4"
              width="18"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                clip-rule="evenodd"
                d="M2 0C3.10457 0 4 0.895431 4 2C4 3.10457 3.10457 4 2 4C0.895431 4 0 3.10457 0 2C0 0.895431 0.895431 0 2 0ZM9 0C10.1046 0 11 0.895431 11 2C11 3.10457 10.1046 4 9 4C7.89543 4 7 3.10457 7 2C7 0.895431 7.89543 0 9 0ZM18 2C18 0.895431 17.1046 0 16 0C14.8954 0 14 0.895431 14 2C14 3.10457 14.8954 4 16 4C17.1046 4 18 3.10457 18 2Z"
                fill="currentColor"
                fill-rule="evenodd"
              />
            </svg>
          </i>
        </div>
      </button>
      
      <!---->
      
    </div>
    
    
  </div>
</div>
`;
